--- sysfsutils-0.4.0/configure.ac.orig	2003-12-17 12:34:17.000000000 +0100
+++ sysfsutils-0.4.0/configure.ac	2003-12-18 20:15:13.347874568 +0100
@@ -10,13 +10,8 @@
 AC_PROG_INSTALL
 AC_PROG_LN_S
 AC_PROG_MAKE_SET
-AC_PROG_RANLIB
 AC_PROG_AWK
 
-# Checks for libraries.
-# FIXME: Replace `main' with a function in `-lsysfs':
-AC_CHECK_LIB([sysfs], [sysfs_read_directory])
-
 # Checks for header files.
 AC_HEADER_DIRENT
 AC_HEADER_STDC
