--- sysfsutils-2.1.1/configure.ac.orig	2023-12-24 11:40:54.621123222 +0100
+++ sysfsutils-2.1.1/configure.ac	2023-12-24 11:41:22.687637839 +0100
@@ -1,5 +1,5 @@
 # Process this file with autoconf to produce a configure script.
-AC_INIT([sysfsutils], [2.1.1], [https://github.com/linux-ras/sysfsutils/issues])
+AC_INIT([sysfsutils], [2.2.0], [https://github.com/linux-ras/sysfsutils/issues])
 AC_CONFIG_SRCDIR([cmd/systool.c])
 AC_CONFIG_MACRO_DIR([m4])
 
--- sysfsutils-2.1.1/lib/Makefile.am.orig	2023-12-24 11:42:40.427216688 +0100
+++ sysfsutils-2.1.1/lib/Makefile.am	2023-12-24 11:43:52.083495160 +0100
@@ -7,6 +7,6 @@ lib_LTLIBRARIES = libsysfs.la
 libsysfs_la_SOURCES = sysfs_utils.c sysfs_attr.c sysfs_class.c dlist.c \
       sysfs_device.c sysfs_driver.c sysfs_bus.c sysfs_module.c sysfs.h
 libsysfs_la_CPPFLAGS = -I$(top_srcdir)/include
-libsysfs_la_LDFLAGS = -version-info 2:1:0
+libsysfs_la_LDFLAGS = -version-info 2:2:0
 EXTRA_CFLAGS = @EXTRA_CLFAGS@
 libsysfs_la_CFLAGS = -Wall -W -Wextra -Wstrict-prototypes $(EXTRA_CLFAGS)
